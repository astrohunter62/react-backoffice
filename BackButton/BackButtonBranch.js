import React from 'react'
import PropTypes from 'prop-types'

import Button from 'material-ui/Button'
import BackIcon from 'material-ui-icons/KeyboardArrowLeft'
import withStyles from 'material-ui/styles/withStyles'

const styles = (theme) => ({
  leftIcon: {
    marginRight: theme.spacing.unit,
  },
})

const BackButtonBranch = ({
  handleBack,
  classes,
}) => (
  <Button onClick={handleBack}>
    <BackIcon className={classes.leftIcon} />
    Back
  </Button>
)

BackButtonBranch.propTypes = {
  handleBack: PropTypes.func.isRequired,
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(BackButtonBranch)
