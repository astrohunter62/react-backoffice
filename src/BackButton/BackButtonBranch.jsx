import React from 'react'
import PropTypes from 'prop-types'

import Button from 'material-ui/Button'
import BackIcon from '@material-ui/icons/KeyboardArrowLeft'
import withStyles from 'material-ui/styles/withStyles'

const styles = theme => ({
  leftIcon: {
    marginRight: theme.spacing.unit,
  },
})

const BackButtonBranch = ({
  onNavigateBack,
  classes,
}) => (
  <Button onClick={onNavigateBack}>
    <BackIcon className={classes.leftIcon} />
    Back
  </Button>
)

BackButtonBranch.propTypes = {
  onNavigateBack: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

export default withStyles(styles)(BackButtonBranch)
