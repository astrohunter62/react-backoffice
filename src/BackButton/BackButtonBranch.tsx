import React from "react";
import { Button, withStyles } from "@material-ui/core";
import BackIcon from "@material-ui/icons/ArrowBackIos";
const styles = (theme: any) => ({
  leftIcon: {
    marginRight: theme.spacing(),
  },
});
type BackButtonBranchProps = {
  onNavigateBack: (...args: any[]) => any;
  classes: {
    [key: string]: string;
  };
};
const BackButtonBranch: React.SFC<BackButtonBranchProps> = ({
  onNavigateBack,
  classes,
}) => (
  <Button onClick={onNavigateBack}>
    <BackIcon className={classes.leftIcon} />
    Back
  </Button>
);
export default withStyles(styles)(BackButtonBranch);
