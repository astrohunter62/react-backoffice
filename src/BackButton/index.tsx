import React, { FunctionComponent } from "react";
import { useHistory } from "react-router-dom";
import { Button } from "@material-ui/core";
import BackIcon from "@material-ui/icons/ArrowBackIos";

type Props = {
  url: string;
};

const BackButton: FunctionComponent<Props> = ({ url }) => {
  const history = useHistory();

  return (
    <Button
      onClick={() => {
        history.push(url);
      }}
      variant="outlined"
      startIcon={<BackIcon />}
    >
      Back
    </Button>
  );
};

export default BackButton;
