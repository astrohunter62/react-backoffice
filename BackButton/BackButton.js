import React from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom'

import BackButtonBranch from './BackButtonBranch'

const withBackButton = (Component) => class extends React.Component {
  constructor() {
    super()

    this.handleBack = this.handleBack.bind(this)
  }

  handleBack() {
    const { url, history } = this.props

    history.push(url)
  }

  render () {
    return (
      <Component
        {...this.props}
        handleBack={this.handleBack}
      />
    )
  }
}

const BackButton = withBackButton(BackButtonBranch)

BackButton.propTypes = {
  url: PropTypes.string.isRequired,
}

export default withRouter(BackButton)
