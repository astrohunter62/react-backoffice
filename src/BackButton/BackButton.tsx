import React from "react";
import { withRouter } from "react-router-dom";
import BackButtonBranch from "./BackButtonBranch";

type WithBackButtonProps = {
  url: string;
  history: {
    [key: string]: any;
  };
};

const withBackButton = (Component: any) =>
  class WithBackButton extends React.Component<WithBackButtonProps, {}> {
    constructor(props: WithBackButtonProps) {
      super(props);
      this.handleBack = this.handleBack.bind(this);
    }
    handleBack() {
      const { url, history } = this.props;
      history.push(url);
    }
    render() {
      return <Component {...this.props} onNavigateBack={this.handleBack} />;
    }
  };

const BackButton = withBackButton(BackButtonBranch);

export default withRouter(BackButton as any) as any;
