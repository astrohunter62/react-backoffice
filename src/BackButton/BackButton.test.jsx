import React from 'react'
import MockRouter from 'react-mock-router'
import { shallow } from 'enzyme'

import BackButton from '.'

describe('Back Button', () => {
  it('renders correctly', () => {
    const tree = shallow((<MockRouter><BackButton url="/" /></MockRouter>))

    expect(tree).toMatchSnapshot()
  })

  it('updates history on click', () => {
    const push = jest.fn()
    const button = shallow(<MockRouter push={push}><BackButton url="/" /></MockRouter>)

    button.find(BackButton).simulate('click')

    expect(push).toHaveBeenLastCalledWith('/')
  })
})
