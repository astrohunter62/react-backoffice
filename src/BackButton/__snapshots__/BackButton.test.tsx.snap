// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back Button renders correctly 1`] = `
<BackButton
  url="/"
/>
`;
